### PR TITLE
[browser] Use Wasm SDK in browser template

### PR DESCRIPTION
**Pull request by maraf**
- Update browser template to use Wasm SDK
- Use Blazor DevServer to run browser template
- Rename mono-config to blazor.boot.json to unify location where to load config from
- Update Wasm.Build.Tests

### CODE REVIEW

File: src/mono/wasm/templates/templates/browser/browser.0.csproj [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
-    <RuntimeIdentifier>browser-wasm</RuntimeIdentifier>
-    <OutputType>Exe</OutputType>
+    <TargetFramework>net8.0</TargetFramework>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
   </PropertyGroup>
 
   <ItemGroup>
-    <WasmExtraFilesToDeploy Include="index.html" />
-    <WasmExtraFilesToDeploy Include="main.js" />
+    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="8.0.0-preview.4.23177.9" PrivateAssets="all" />
```
Comment (maraf) [inline src/mono/wasm/templates/templates/browser/browser.0.csproj:8]: @radical @lewing Is there a way to manage this version?

File: src/tasks/Microsoft.NET.Sdk.WebAssembly.Pack.Tasks/ComputeWasmPublishAssets.cs [hunk 2]
```diff
@@ -197,11 +197,16 @@ public override bool Execute()
             {
                 var aotDotNetJs = WasmAotAssets.SingleOrDefault(a => $"{a.GetMetadata("FileName")}{a.GetMetadata("Extension")}" == "dotnet.js");
                 ITaskItem newDotNetJs = null;
-                if (aotDotNetJs != null && FingerprintDotNetJs)
+                if (aotDotNetJs != null)
```
Comment (maraf) [inline src/tasks/Microsoft.NET.Sdk.WebAssembly.Pack.Tasks/ComputeWasmPublishAssets.cs:200]: This is not exactly related to the rest of the PR, but fixes using relinked dotnet.js when fingerprinting is off

File: src/mono/wasm/templates/templates/browser/browser.0.csproj (hunk 1 above)
Comment (lambdageek) [reply src/mono/wasm/templates/templates/browser/browser.0.csproj:8]: aspnetcore uses .csproj.in template files and fills them in with a generated version. https://github.com/dotnet/aspnetcore/blob/015ceab6aeec700fbfb71cdfe9ff8fb51831d555/src/ProjectTemplates/Web.ProjectTemplates/Microsoft.DotNet.Web.ProjectTemplates.csproj

**Comment by maraf** [comment]
Closed in favor of https://github.com/dotnet/runtime/pull/85498

**Pull request closed without merging**